e2e/packages/static-theme/src/index.tsx

```diff
@@ -0,0 +1,142 @@
+import { css, styled, Global } from "frontity";
+
+const globalStyles = css`
+  :root {
+    --blue-1: #79aad7;
+    --blue-2: #aecae8;
+    --dark-1: #151515;
+    --dark-2: #444;
+    --light-1: #eee;
+  }
+
+  html,
+  body {
+    margin: 0;
+    padding: 0;
+    width: 100%;
+    height: 100%;
+    display: flex;
+    flex-direction: column;
+    font-family: sans-serif;
+    font-size: 1em;
+    background: var(--dark-1);
+    color: var(--light-1);
+  }
+
+  * {
+    box-sizing: border-box;
+  }
+`;
+
+/**
+ * The colors.
+ */
+type Colors = "blue-1" | "blue-2" | "dark-1" | "dark-2" | "light-1";
+
+/**
+ * The custom styled Box.
+ */
+interface BoxProps {
+  /**
+   * Padding value.
+   */
+  p?: number;
+
+  /**
+   * Margin value.
+   */
+  m?: number;
+
+  /**
+   * Background color.
+   */
+  bg?: Colors;
+
+  /**
+   * The text color.
+   */
+  color?: Colors;
+
+  /**
+   * The font size.
+   */
+  size?: number;
+
+  /**
+   * The border radius.
+   */
+  r?: number;
+}
+
+const Box = styled.div<BoxProps>((props) => [
+  {
+    display: "flex",
+    flexDirection: "column",
+    margin: 0,
+    padding: 0,
+  },
+  props.p && { padding: props.p + "em" },
+  props.m && { margin: props.m + "em" },
+  props.bg && { background: `var(--${props.bg})` },
+  props.color && { color: `var(--${props.color})` },
+  props.size && { fontSize: props.size + "em" },
+  props.r && { borderRadius: props.r + "em" },
+]);
```

Comment:
Some of the props like `color` or `size` are ending up in the DOM and it's disallowed in AMP (at least according to https://validator.ampproject.org/). Have a look:

https://www.loom.com/share/ad5fbecd48644b549f2b070820a064b8

Reply:
Oh, I see. And since we can't control the end-user usages we might at least control our tests and results to make sure their compatible. Is this your line of thought?

Reply:
Yup, that was my thinking basically 🙂 I thought that `emotion` should be filtering out the props passed to styled components that are not part of the HTML spec but maybe that's not the case.